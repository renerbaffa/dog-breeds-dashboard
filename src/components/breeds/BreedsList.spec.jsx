import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import ConnectedBreedsList, {
  BreedsList,
} from './BreedsList';

import BREEDS from '../../mocks/Breeds';
import normalizeBreeds from '../../normalizers/breeds';

const NORMALIZED_BREEDS = normalizeBreeds(BREEDS);

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('<BreedsList />', () => {
  let component;
  let onFetchBreeds;
  let wrapper;

  beforeEach(() => {
    onFetchBreeds = jest.fn();
    component = (
      <BreedsList
        breeds={NORMALIZED_BREEDS}
        onFetchBreeds={onFetchBreeds}
      />
    );
    wrapper = shallow(component);
  });

  it('should render correctly', () => {
    const breedsList = renderer.create(component).toJSON();
    expect(breedsList).toMatchSnapshot();
  });

  it('should call onFetchBreeds on componentDidMount', () => {
    onFetchBreeds = jest.fn();
    wrapper.setProps({ onFetchBreeds });
    expect(onFetchBreeds).not.toHaveBeenCalled();
    wrapper.instance().componentDidMount();
    expect(onFetchBreeds).toHaveBeenCalled();
  });

  it('should inject breeds from store into component', () => {
    const store = mockStore({ breeds: NORMALIZED_BREEDS });
    wrapper = shallow(<ConnectedBreedsList
      onFetchBreeds={onFetchBreeds}
      store={store}
    />);
  });

  it('should reach 100% of coverage', () => {
    BreedsList.defaultProps.onFetchBreeds();
  });
});
