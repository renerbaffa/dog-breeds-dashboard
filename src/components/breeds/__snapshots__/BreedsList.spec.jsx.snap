// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreedsList /> should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <div>
      American (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Australian (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Bedlington (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Border (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Dandie (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Fox (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Irish (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Kerryblue (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Lakeland (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Norfolk (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Norwich (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Patterdale (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Scottish (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Sealyham (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Silky (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Tibetan (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Toy (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Westhighland (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Wheaten (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Yorkshire (Terrier)
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Vizsla
    </div>
  </div>
  <div
    className="container"
  >
    <div>
      Weimaraner
    </div>
  </div>
</div>
`;
